arrows/mvg/applets/init_cameras_landmarks.cxx

```diff
@@ -0,0 +1,644 @@
+  }
+  else
+  {
+    std::string path = config->get_value<std::string>("input_tracks_file");
+    if ( ! ST::FileExists( kwiver::vital::path_t(path), true ) )
+    {
+      KWIVER_CONFIG_FAIL("input_tracks_file path, " << path
+                         << ", does not exist or is not a regular file");
+    }
+  }
+
+  if (!config->has_value("output_cameras_directory") ||
+    config->get_value<std::string>("output_cameras_directory") == "" )
+  {
+    KWIVER_CONFIG_FAIL("Config needs value output_cameras_directory");
+  }
+  else if ( ! ST::FileIsDirectory(
+    config->get_value<kwiver::vital::path_t>("output_cameras_directory") ) )
+  {
+    KWIVER_CONFIG_FAIL("output_cameras_director is not a valid directory");
+  }
+
+  if (!config->has_value("output_landmarks_filename") ||
+    config->get_value<std::string>("output_landmarks_filename") == "" )
+  {
+    KWIVER_CONFIG_FAIL("Config needs value output_landmarks_filename");
+  }
+  else if ( ! ST::FileIsDirectory( ST::CollapseFullPath( ST::GetFilenamePath(
+    config->get_value<kwiver::vital::path_t>("output_landmarks_filename") ))))
+  {
+    KWIVER_CONFIG_FAIL("output_landmarks_filename is not in a valid directory");
+  }
+
+  {
+    kwiver::vital::path_t out_landmarks_path =
+      config->get_value<kwiver::vital::path_t>("output_landmarks_filename");
+
+    // verify that we can open the output file for writing
+    // so that we don't find a problem only after spending
+    // hours of computation time.
+    std::ofstream ofs(out_landmarks_path.c_str());
+    if (!ofs)
+    {
+      KWIVER_CONFIG_FAIL("Could not open landmark file for writing: \""
+                         + out_landmarks_path + "\"");
+    }
+    ofs.close();
+  }
+
+#undef KWIVER_CONFIG_FAIL
+
+  return config_valid;
+}
+} // end namespace
+
+class init_cameras_landmarks::priv
+{
+public:
+  priv()
+  : camera_map_ptr(nullptr),
+    landmark_map_ptr(nullptr),
+    feature_track_set_ptr(nullptr),
+    sfm_constraint_ptr(nullptr),
+    config(nullptr),
+    num_frames(0)
+  {}
+
+  camera_map_sptr camera_map_ptr;
+  landmark_map_sptr landmark_map_ptr;
+  feature_track_set_sptr feature_track_set_ptr;
+  sfm_constraints_sptr sfm_constraint_ptr;
+  initialize_cameras_landmarks_sptr algorithm;
+  kwiver::vital::config_block_sptr config;
+  size_t num_frames;
+  std::string video_name;
+
+  enum commandline_mode {SUCCESS, HELP, WRITE, FAIL};
+
+  commandline_mode process_command_line(cxxopts::ParseResult& cmd_args)
+  {
+    static std::string opt_config;
+    static std::string opt_out_config;
+
+    if ( cmd_args["help"].as<bool>() )
+    {
+      return HELP;
+    }
+    if ( cmd_args.count("config") )
+    {
+      opt_config = cmd_args["config"].as<std::string>();
+    }
+    if ( cmd_args.count("output-config") > 0 )
+    {
+      opt_out_config = cmd_args["output-config"].as<std::string>();
+    }
+
+    // Set up top level configuration w/ defaults where applicable.
+    config = default_config();
```

Comment:
See #1102 for and example of how to replace this function with a pointer to a default config file that we can put in `kwiver/config/applets/`.  Of course that branch will need to land first